packaging/macos/apphost-mono.c

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2007-2022 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+
+//
+// A custom apphost is required (instead of just invoking `mono OpenRA.dll ...` directly)
+// because macOS will only properly associate dock icons and tooltips to windows that are
+// created by a process in the Contents/MacOS directory (not subdirectories).
+//
+
+#include <dlfcn.h>
+#include <stdio.h>
+#include <sys/resource.h>
+
+typedef int (* mono_main)(int argc, char **argv);
+
+int main(int argc, char **argv)
+{
+	struct rlimit limit;
+	if (getrlimit(RLIMIT_NOFILE, &limit) == 0 && limit.rlim_cur < 1024)
+	{
+		limit.rlim_cur = limit.rlim_max < 1024 ? limit.rlim_max : 1024;
+		setrlimit(RLIMIT_NOFILE, &limit);
```

Comment:
I don't actually know. This was originally copied from https://github.com/mono/monodevelop/blob/main/main/build/MacOSX/monostub.mm in our first native launcher and has been there ever since.

Reply:
Ok weird. I couldn't find any explanation on why that was added there either. Maybe we can add a comment here (saying where this comes from and that we don't know if it's actually necessary) and leave it as is?

Reply:
Tracked back to https://github.com/mono/monodevelop/commit/030ef63e0ccb1d3faac79ea24832fa286003ab13 which isn't very helpful.